core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -951,15 +951,22 @@ class KafkaApis(val requestChannel: RequestChannel,
               new OffsetFetchResponse(Errors.NONE, authorizedPartitionData.asJava, header.apiVersion)
             }
           } else {
-            val (authorizedPartitions, unauthorizedPartitions) = offsetFetchRequest.partitions.asScala
-              .partition(authorizeTopicDescribe)
-            val (error, authorizedPartitionData) = coordinator.handleFetchOffsets(offsetFetchRequest.groupId,
-              Some(authorizedPartitions))
+            val (authorizedPartitions, unauthorizedPartitions) =
+              offsetFetchRequest.partitions.asScala.partition(authorizeTopicDescribe)
+            val (error, authorizedPartitionData) =
+              coordinator.handleFetchOffsets(offsetFetchRequest.groupId, Some(authorizedPartitions))
             if (error != Errors.NONE)
               offsetFetchRequest.getErrorResponse(error)
             else {
+              val authorizedAndValidatedPartitionData = authorizedPartitionData.map {
+                case (topicPartition, offsetInfo) =>
+                  if (metadataCache.getPartitionInfo(topicPartition.topic, topicPartition.partition).isEmpty)
```

Comment:
@vahidhashemian I think the motivation of the KIP-88 is to allow broker to specify response-level-error when there is no partition in the response. Does the KIP actually specify that where we should specify error when response contains at least one partition information? If not, then there is no concern with modifying behavior here.

It seems confusing to specify not-null error at both request level and partition level. We probably want to adopt the same practice used for LeaderAndIsrResponse. That is broker should only specify not-null partition-level error with response-level-error = null. And client should only read partition-level-error when request-level-error is null. In other words, the most specific result wins.

Thus, if the response contains at least one partition, I think we can set response-level-error to none and provide appropriate partition-level-error.